include/bredis/impl/async_connection.ipp

```diff
@@ -0,0 +1,136 @@
+//
+//
+// Copyright (c) 2017 Ivan Baidakou (basiliscos) (the dot dmol at gmail dot com)
+//
+// Distributed under the MIT Software License
+//
+#pragma once
+
+#include <algorithm>
+#include <cassert>
+#include <type_traits>
+
+#include "common.ipp"
+
+namespace bredis {
+
+template <typename AsyncStream>
+template <typename WriteCallback>
+void AsyncConnection<AsyncStream>::async_write(const command_wrapper_t &command,
+                                               WriteCallback write_callback) {
+    namespace asio = boost::asio;
+    namespace sys = boost::system;
+
+    auto str = std::make_shared<std::string>(
+        boost::apply_visitor(command_serializer_visitor(), command));
+    auto str_ptr = str.get();
+    BREDIS_LOG_DEBUG("async_write >> " << str_ptr->c_str());
```

Comment:
There's a different way to log which doesn't require the macro or explicit statements. You could simply provide a logging wrapper which wraps the socket, and instantiate the class with that type as the next layer type.